﻿﻿﻿# ﻿﻿﻿﻿﻿﻿﻿﻿﻿﻿BCB546x_Unix_Assignment### Getting into my files using ls and cd commands  ` ls `BCB546X-Fall2018  bds-files  in_class`cd BCB546X-Fall2018/``ls`assignments  course-files  documents  README.md`cd assignments/``ls`README.md  UNIX_Assignment`cd UNIX_Assignment/``ls`fang_et_al_genotypes.txt  transpose.awk  UNIX_Assignment.pdf snp_position.txt          UNIX_Assignment.md### -Inspecting file size using the du commands`du -h fang_et_al_genotypes.txt`11M     fang_et_al_genotypes.txt`du -h snp_position.txt`84K     snp_position.txt#This is how to expect the file###-Inspecting the number of columns in a file with awk and cut`awk -F "\t" '{print NF; exit}' fang_et_al_genotypes.txt`986`awk -F "\t" '{print NF; exit}' snp_position.txt`15`awk -F "\t" '{print NF; exit}' fang`986`cut -f 1 fang_et_al_genotypes.txt| wc -l`2783### -How to look at the file using uniq  comand `cut -f 3 fang_et_al_genotypes.txt |sort| uniq -c`      1 Group     22 TRIPS     15 ZDIPL     17 ZLUXR     10 ZMHUE    290 ZMMIL   1256 ZMMLR     27 ZMMMR    900 ZMPBA     41 ZMPIL     34 ZMPJA     75 ZMXCH     69 ZMXCP      6 ZMXIL      7 ZMXNO      4 ZMXNT      9 ZPERR### -How to get the files ZMMIL ZMMMR ZMMLR ```grep 'ZMM' fang_et_al_genotypes.txt > maize_genotype.txtls```fang_et_al_genotypes.txt  transpose.awk             UNIX_Assignment.pdfmaize_genotype.txt        transposed_genotypes.txtsnp_position.txt          UNIX_Assignment.md#- you can see here that i created  a new maze genotype folder using the grep comand #create a new file the using the head command `$head -n 1 fang_et_al_genotypes.txt > head.txt`# - this provides you with the header of you file ## Now make a new that includes both the head and maize files using `cat` ```cat header.txt maize_genotype.txt > head_maize_genotype.txt```# transpose your file form above `$ awk -f transpose.awk head_maize_genotype.txt > transposed_head_maize_genotypes.txt`## you can inspect the file```$cut -f 1 transposed_head_maize_genotypes.txt | wc```    986     986   10847`$ cut -f 1 transposed_head_maize_genotypes.txt`#- Piping it to a `wc` comand will give you the word count of each collum-  If you do not pipe it you will see all the data that are in your new transposed folder- This is a good way to check your file # Sort the trasposed head maize genotype'tail -n +4 transposed_head_maize_genotypes.txt| sort -k1,1 > sorted_transposed_head_maize_genotypes.txt''cut -f 1 sorted_transposed_header_maize_genotypes.txt | wc'    983     983   10824- I did this cut comand in order to take the unsusfull information such as the sample id # sort the snp_position.txt file - check if your file is sorted `sort -c snp_position.txt` ```head -n 1 snp_position.txt > snp_head.txttail -n +2 snp_position.txt | sort -k1,1 > sorted_snp_position.txtcut -f 1 sorted_snp_position.txt```# Join sorted snp files and sorted transposed head_maize_genotype data using the `join` command```join -t $'\t' -1 1 -2 1 sorted_snp_position.txt sorted_transposed_head_maize_genotypes.txt > joined_maize.txt```we need to add the snp headers  ```cat snp_head.txt joined_maize.txt > joined_head_maize_genotype_snp.txtcat joined_head_maize_genotype_snp.txt| cut -f 1-16| head```- we can de here that their is some information that wee need to take out `$ awk -F "\t" '{print NF; exit}' joined_head_maize_genotype_snp.txt`15#- this tells you how many columns their is, we need to take out the 2nd column  and keep the rest  'cut -f 1,3,4,5,6,7,8,9,10,11,12,13,14,15- joined_head_maize_genotype_snp.txt | head | cut -f 1-10''cut -f 1,3,4,5,6,7,8,9,10,11,12,13,14,15- joined_head_maize_genotype_snp.txt > reordered_maize_genotype_snp.txt'#look at the data `cut -f 2 reordered_maize_genotype_snp.txt |sort -n | uniq -c` 1 Chromosome      6 multiple     27 unknown    155 1    127 2    107 3     91 4    122 5     76 6     97 7     62 8     60 9     53 10#creating files for the chromosomes 'awk '$2 == 1' reordered_maize_genotype_snp.txt > maize_chromosome_1.txt'#looking in the files `cut -f 2 maize_chromosome_1.txt| sort| uniq -c`    155 1'cut -f 1- maize_chromosome_1.txt | head | cut -f 1-3'an1.4   1       240498509id1.3   1       238902078'awk '$2 == 1' reordered_maize_genotype_snp.txt > maize_chromosome_1.txt'`awk '$2 == 2' reordered_maize_genotype_snp.txt > maize_chromosome_2.txt``awk '$2 == 3' reordered_maize_genotype_snp.txt > maize_chromosome_3.txt``awk '$2 == 4' reordered_maize_genotype_snp.txt > maize_chromosome_4.txt``awk '$2 == 5’ reordered_maize_genotype_snp.txt > maize_chromosome_5.txt``awk '$2 == 6' reordered_maize_genotype_snp.txt > maize_chromosome_6.txt``awk '$2 == 7’ reordered_maize_genotype_snp.txt > maize_chromosome_7.txt``awk '$2 == 8’ reordered_maize_genotype_snp.txt > maize_chromosome_8.txt``awk '$2 == 9’ reordered_maize_genotype_snp.txt > maize_chromosome_9.txt``awk '$2 == 10' reordered_maize_genotype_snp.txt > maize_chromosome_10.txt`#looking inside the files`cut -f 2 maize_chromosome_10.txt| sort| uniq -c`     53 10`cut -f 2 maize_chromosome_1.txt| sort| uniq -c`    155 1`cut -f 2 maize_chromosome_2.txt| sort| uniq -c`    127 2`cut -f 2 maize_chromosome_3.txt| sort| uniq -c`    107 3`cut -f 2 maize_chromosome_4.txt| sort| uniq -c`     91 4`cut -f 2 maize_chromosome_5.txt| sort| uniq -c`    122 5`cut -f 2 maize_chromosome_6.txt| sort| uniq -c`     76 6`cut -f 2 maize_chromosome_7.txt| sort| uniq -c`     97 7`cut -f 2 maize_chromosome_8.txt| sort| uniq -c`     62 8`cut -f 2 maize_chromosome_9.txt| sort| uniq -c`     60 9`cut -f 2 maize_chromosome_10.txt| sort| uniq -c`     53 10# we see that our chromosome file is not sorted   `cat maize_chromosome_1.txt|head |cut -f 1-6|column -t`an1.4        1  240498509  C/G  C/G  C/Cid1.3        1  238902078  C/T  C/T  C/CPZA00010.5   1  119703101  C/C  A/C  A/APZA00013.10  1  263276247  T/T  ?/?  A/APZA00013.11  1  263276005  C/C  C/C  C/CPZA00013.9   1  263276335  T/T  T/T  T/TPZA00017.1   1  35620725   A/A  G/G  A/APZA00050.9   1  295459549  A/A  A/A  A/GPZA00081.17  1  45675053   C/C  C/C  C/CPZA00098.14  1  161472433  C/T  C/C  C/C#sorting in increasing order for maize chromosome `sort -k3,3n maize_chromosome_1.txt >increasing_maize_chromosome_1.txt``sort -k3,3n maize_chromosome_2.txt >increasing_maize_chromosome_2.txt``sort -k3,3n maize_chromosome_3.txt >increasing_maize_chromosome_3.txt``sort -k3,3n maize_chromosome_4.txt >increasing_maize_chromosome_4.txt``sort -k3,3n maize_chromosome_5.txt >increasing_maize_chromosome_5.txt``sort -k3,3n maize_chromosome_6.txt >increasing_maize_chromosome_6.txt``sort -k3,3n maize_chromosome_7.txt >increasing_maize_chromosome_7.txt``sort -k3,3n maize_chromosome_8.txt >increasing_maize_chromosome_8.txt``sort -k3,3n maize_chromosome_9.txt >increasing_maize_chromosome_9.txt``sort -k3,3n maize_chromosome_10.txt >increasing_maize_chromosome_10.txt`#To check if the sort is done correctly`cat increasing_maize_chromosome_1.txt|head |cut -f 1-6|column -t`PZB00859.1   1  157104   C/C  C/C  A/CPZA02962.13  1  3205252  T/T  T/T  T/TPZA00393.1   1  4175293  C/T  C/C  T/TPZA00393.4   1  4175573  G/G  G/G  G/GPZA02869.8   1  4429897  C/C  C/C  ?/?- we can see here that there are ? for unknown positions# to sort from decreasing `nr` command and change the ? to - we use the `sed` command `sort -k3,3nr maize_chromosome_1.txt| sed 's/?/-/g'| head| cut -f 1-10`PZA00230.5      1       298412984       G/G     G/G     -/-     C/C     C/C     C/C     C/GPZA00477.11     1       298082627       G/G     G/G     G/G     A/G     A/G     A/A     A/GPZA00477.9      1       298082534       C/C     C/C     C/C     C/C     C/C     C/C     C/CPZA00477.10     1       298082504       G/G     G/G     G/G     G/G     G/G     G/G     G/GPZA00477.5      1       298082468       T/T     T/T     T/T     T/T     T/T     T/T     T/TPZA00243.27     1       295771152       A/A     A/A     A/A     A/A     A/A     A/A     A/APZA00050.9      1       295459549       A/A     A/A     A/G     A/A     A/A     A/A     A/APZA00623.2      1       293632755       C/G     -/-     G/G     G/G     G/G     G/G     G/GPZA00235.8      1       292728923       G/G     G/G     G/G     A/G     A/G     G/G     G/GPZA00235.6      1       292728854       G/T     G/G     G/G     G/T     T/T     G/T     G/G`sort -k3,3nr maize_chromosome_1.txt| sed 's/?/-/g' > decreasing_maize_chromosome_1.txt``sort -k3,3nr maize_chromosome_2.txt| sed 's/?/-/g' > decreasing_maize_chromosome_2.txt``sort -k3,3nr maize_chromosome_3.txt| sed 's/?/-/g' > decreasing_maize_chromosome_3.txt``sort -k3,3nr maize_chromosome_4.txt| sed 's/?/-/g' > decreasing_maize_chromosome_4.txt``sort -k3,3nr maize_chromosome_5.txt| sed 's/?/-/g' > decreasing_maize_chromosome_5.txt``sort -k3,3nr maize_chromosome_6.txt| sed 's/?/-/g' > decreasing_maize_chromosome_6.txt``sort -k3,3nr maize_chromosome_7.txt| sed 's/?/-/g' > decreasing_maize_chromosome_7.txt``sort -k3,3nr maize_chromosome_8.txt| sed 's/?/-/g' > decreasing_maize_chromosome_8.txt``sort -k3,3nr maize_chromosome_9.txt| sed 's/?/-/g' > decreasing_maize_chromosome_9.txt``sort -k3,3nr maize_chromosome_10.txt| sed 's/?/-/g' > decreasing_maize_chromosome_10.txt``cat decreasing_maize_chromosome_1.txt|head | cut -f 1-6| column -t`PZA00230.5   1  298412984  G/G  G/G  -/-PZA00477.11  1  298082627  G/G  G/G  G/GPZA00477.9   1  298082534  C/C  C/C  C/CPZA00477.10  1  298082504  G/G  G/G  G/GPZA00477.5   1  298082468  T/T  T/T  T/TPZA00243.27  1  295771152  A/A  A/A  A/APZA00050.9   1  295459549  A/A  A/A  A/GPZA00623.2   1  293632755  C/G  -/-  G/GPZA00235.8   1  292728923  G/G  G/G  G/GPZA00235.6   1  292728854  G/T  G/G  G/G###add the head files `head -n 1 reordered_maize_genotype_snp.txt > head_maize_chromosome.txt`#-this will create a head file `cat head_maize_chromosome.txt decreasing_maize_chromosome_1.txt > decreasing_maize_head_chromosome_1.txt` cut -f 1-5 decreasing_maize_head_chromosome_1.txt | column -t| head -n 5SNP_ID       Chromosome  PositionPZA00230.5   1           298412984PZA00477.11  1           298082627PZA00477.9   1           298082534PZA00477.10  1           298082504`cat head_maize_chromosome.txt decreasing_maize_chromosome_2.txt > decreasing_maize_head_chromosome_2.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_3.txt > decreasing_maize_head_chromosome_3.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_4.txt > decreasing_maize_head_chromosome_4.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_5.txt > decreasing_maize_head_chromosome_5.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_6.txt > decreasing_maize_head_chromosome_6.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_7.txt > decreasing_maize_head_chromosome_7.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_8.txt > decreasing_maize_head_chromosome_8.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_9.txt > decreasing_maize_head_chromosome_9.txt``cat head_maize_chromosome.txt decreasing_maize_chromosome_10.txt > decreasing_maize_head_chromosome_10.txt`#-and increase `cat head_maize_chromosome.txt increasing_maize_chromosome_1.txt > increasing_maize_head_chromosome_1.txt`#teosinte 