﻿﻿﻿﻿﻿﻿﻿﻿﻿﻿﻿# ﻿﻿﻿﻿﻿﻿﻿﻿﻿﻿BCB546x_Unix_Assignment### Getting into my files using ls and cd commands  `cd assignments/``ls`README.md  UNIX_Assignment`cd UNIX_Assignment/`### -Inspecting file size using the du commands`du -h fang_et_al_genotypes.txt`11M     fang_et_al_genotypes.txt`du -h snp_position.txt`84K     snp_position.txt###-Inspecting the number of columns in a file with awk and cut`awk -F "\t" '{print NF; exit}' fang_et_al_genotypes.txt`986`awk -F "\t" '{print NF; exit}' snp_position.txt`15`awk -F "\t" '{print NF; exit}' fang`986`cut -f 1 fang_et_al_genotypes.txt| wc -l`2783### -How to look at the file using uniq  comand `cut -f 3 fang_et_al_genotypes.txt |sort| uniq -c`      1 Group     22 TRIPS     15 ZDIPL     17 ZLUXR     10 ZMHUE    290 ZMMIL   1256 ZMMLR     27 ZMMMR    900 ZMPBA     41 ZMPIL     34 ZMPJA     75 ZMXCH     69 ZMXCP      6 ZMXIL      7 ZMXNO      4 ZMXNT      9 ZPERR### How to get the files ZMMIL ZMMMR ZMMLR `grep 'ZMM' fang_et_al_genotypes.txt > maize_genotype.txtls`#- you can see here that i created  a new maze genotype folder using the grep comand # create a new file the using the head command `head -n 1 fang_et_al_genotypes.txt > head.txt`# - this provides you with the header of you file ## Creating file that includes both the head and maize files using `cat` `cat head.txt maize_genotype.txt > head_maize_genotype.txt`# transpose your file form above `awk -f transpose.awk head_maize_genotype.txt > transposed_head_maize_genotypes.txt`## you can inspect the file`cut -f 1 transposed_head_maize_genotypes.txt | wc`    986     986   10847`$ cut -f 1 transposed_head_maize_genotypes.txt`# Sort the file'tail -n +4 transposed_head_maize_genotypes.txt| sort -k1,1 > sorted_transposed_head_maize_genotypes.txt''cut -f 1 sorted_transposed_head_maize_genotypes.txt | wc'   # check if your file is sorted `sort -c snp_position.txt` `head -n 1 snp_position.txt > snp_head.txt``tail -n +2 snp_position.txt | sort -k1,1 > sorted_snp_position.txt``cut -f 1 sorted_snp_position.txt`# Join sorted snp files and sorted transposed head_maize_genotype data using the `join` command`join -t $'\t' -1 1 -2 1 sorted_snp_position.txt sorted_transposed_head_maize_genotypes.txt > joined_maize.txt`# add the snp headers  `cat snp_head.txt joined_maize.txt > joined_head_maize_genotype_snp.txt``cat joined_head_maize_genotype_snp.txt| cut -f 1-16| head`# word count the lines for all the documents wc -l snp_position.txt sorted_snp_position.txt 984 snp_position.txt   983 sorted_snp_position.txt  1967 total- look at how many columns are in the file `$ awk -F "\t" '{print NF; exit}' joined_head_maize_genotype_snp.txt`15# cut out the columns that we need`cut -f 1,3,4,16- joined_head_maize_genotype_snp.txt | head | cut -f 1-10``cut -f 1,3,4,16- joined_head_maize_genotype_snp.txt > reordered_maize_genotype_snp.txt`# look at the data `cut -f 2 reordered_maize_genotype_snp.txt |sort -n | uniq -c` 1 Chromosome      6 multiple     27 unknown    155 1    127 2    107 3     91 4    122 5     76 6     97 7     62 8     60 9     53 10#creating files for the chromosomes `awk '$2 == 1' reordered_maize_genotype_snp.txt > maize_chromosome_1.txt`#looking in the files `cut -f 2 maize_chromosome_1.txt| sort| uniq -c``cut -f 1- maize_chromosome_1.txt | head | cut -f 1-3`# looking inside the files`cut -f 2 maize_chromosome_10.txt| sort| uniq -c`  # we see that our chromosome file is not sorted   `cat maize_chromosome_1.txt|head |cut -f 1-6|column -t`#sorting in increasing order for maize chromosome `sort -k3,3n maize_chromosome_1.txt > sorted_maize_chromosome_1.txt`#To check if the sort is done correctly`cat sorted_maize_chromosome_1.txt|head |cut -f 1-6|column -t`# to sort from decreasing `nr` command and change the ? to - we use the `sed` command `sort -k3,3nr maize_chromosome_1.txt| sed 's/?/-/g'| head| cut -f 1-10``sort -k3,3nr maize_chromosome_1.txt| sed 's/?/-/g' > reverse_maize_chromosome_1.txt``cat reverse_maize_chromosome_1.txt|head | cut -f 1-6| column -t`###add the head files `head -n 1 reordered_maize_genotype_snp.txt > head_maize_chromosome.txt`# this will create a head+ chromosome file `cat head_maize_chromosome.txt reverse_maize_chromosome_1.txt > decreasing_maize_head_chromosome_1.txt`# And increase `cat head_maize_chromosome.txt sorted_maize_chromosome_1.txt > increasing_maize_head_chromosome_1.txt`# take out the multiple and unknown using grep`grep -vE "unknown|multiple" increasing_teosinte_chromosome_1.txt > teosinte_increasing_chromosome_1.txt` # create your unknown and multiple site file `grep "unknown" reordered_maize_genotype_snp.txt > unknown_maize_chromosome.txt``grep "multiple" reordered_maize_genotype_snp.txt > multiple_maize_chromosome.txt`# Teosinte ### grep the zmp files for your teosinte  `grep 'ZMP' fang_et_al_genotypes.txt > teosinte_genotype.txt`### add the header `cat head.txt teosinte_genotype.txt > head_teosinte_genotype.txt`### transpose the data awk -f transpose.awk head_teosinte_genotype.txt > transposed_head_teosinte_genotypes.txt`cut -f 1 transposed_head_teosinte_genotypes.txt | wc`    986     986   10847` cut -f 1 transposed_head_teosinte_genotypes.txt``tail -n +4 transposed_head_teosinte_genotypes.txt | sort -k1,1 > sorted_transposed_head_teosinte_genotypes.txt`# Join the sorted snp and the sorted teosinte `join -t $'\t' -1 1 -2 1 sorted_snp_position.txt sorted_transposed_head_teosinte_genotypes.txt > joined_teosinte.txt`### Add the snp headers`cat snp_head.txt joined_teosinte.txt > joined_head_teosinte_genotype_snp.txt``cat joined_head_teosinte_genotype_snp.txt| cut -f 1-16| head`# reorder your colums`cut -f 1,3,4,16- joined_header_teosinte_genotype_snp.txt | head | cut -f 1-10| column`` cut -f 1,3,4,16- joined_header_teosinte_genotype_snp.txt > reordered_teosinte_genotype_snp.txt`# Create your chromosomes`awk '$2 == 2' reordered_teosinte_genotype_snp.txt > teosinte_chromosome_2.txt` # getting unknown and multiple sites `grep "unknown" reordered_teosinte_genotype_snp.txt > unknown_teosinte_chromosome.txt``grep "multiple" reordered_teosinte_genotype_snp.txt > multiple_teosinte_chromosome.txt`### Increasing position sort of teosinte chromosome SNPs`sort -k3,3n teosinte_chromosome_1.txt|head| cut -f 1-3``sort -k3,3n teosinte_chromosome_1.txt >sorted_teosinte_chromosome_1.txt``cat sorted_teosinte_chromosome_1.txt| head |cut -f 1-6|column -t``sort -k3,3n teosinte_chromosome_1.txt > sorted_teosinte_chromosome_1.txt`### Decreasing position order and replace ‘?’ with ‘-‘`sort -k3,3nr teosinte_chromosome_10.txt | head| cut -f 1-10``sort -k3,3nr teosinte_chromosome_10.txt| sed 's/?/-/g'| head| cut -f 1-10```sort -k3,3nr teosinte_chromosome_10.txt| sed 's/?/-/g'> reverse_teosinte_chromosome_10.txt`## Add the headers`cat head_maize_chromosome.txt sorted_teosinte_chromosome_1.txt > increasing_teosinte_chromosome_1.txt``cat head_maize_chromosome.txt reverse_teosinte_chromosome_1.txt > decreasing_teosinte_chromosome_1.txt` 